src/absil/il.fs

```diff
@@ -95,12 +93,10 @@ let memoizeNamespaceArrayTable =
     Dictionary<string,string[]>(10)
 
 let splitNamespaceToArray nm =
-    let mutable res = Unchecked.defaultof<_>
-    let ok = memoizeNamespaceArrayTable.TryGetValue(nm,&res)
+    let ok, res = memoizeNamespaceArrayTable.TryGetValue(nm)
     if ok then res else
     let x = Array.ofList (splitNamespace nm)
-    (memoizeNamespaceArrayTable.[nm] <- x; x)
-
+    lock memoizeNamespaceArrayTable (fun () -> memoizeNamespaceArrayTable.[nm] <- x; x)
```

Comment:
If we are locking on `memoizeNamespaceArrayTable`, it should be made private so that other components can't also lock on it.  Switching to ConcurrentDictionary of course resolves that problem.